src/python/pants/python/python_setup.py

```diff
@@ -204,15 +209,18 @@ def artifact_cache_dir(self):
     def scratch_dir(self):
         return os.path.join(self.get_options().pants_workdir, *self.options_scope.split("."))
 
-    def compatibility_or_constraints(self, compatibility):
-        """Return either the given compatibility, or the interpreter constraints. If interpreter
-        constraints are supplied by the CLI flag, return those only.
+    def compatibility_or_default(self, compatibility):
```

Comment:
Type hints appreciated. I believe this is `Optional[List[str]]` and returns `List[str]`? Likely it should return a tuple to mirror `compatibility_for_target`.